src/framework/components/anim/controller.js

```diff
@@ -485,7 +485,9 @@ Object.assign(AnimController.prototype, {
             var interpolatedTime = this._currTransitionTime / this._totalTransitionTime;
             for (i = 0; i < this._transitionPreviousStates.length; i++) {
                 // interpolate the weights of the most recent previous state and all other previous states based on the progress through the previous transition
-                if (i !== this._transitionPreviousStates.length - 1) {
+                if (!this._isTransitioning) {
+                    this._transitionPreviousStates[i].weight = 1.0;
```

Comment:
Just curious, but why use 1.0 instead of 1? In JS, aren't they identical?

Reply:
I've got into the habit of distinguishing ints and floats so I can notice what that var is being used as. They are identical.

Reply:
Fair enough. It's not a big deal. 😄